[18.0][MIG] account_payment_sale

alexis-via opened the pull request:
I'll publish the PR for account_payment_base_oca

Commit:
Add 4 new modules to handle payment type and bank accounts:
- account_payment_partner
- account_payment_sale
- account_payment_sale_stock
- account_payment_purchase

Filter the selection of invoices per payment type.
Add active field on payment.mode and payment.mode.type.
Add menu entry for Payment Types.

pedrobaeza committed:
[MIG] account_payment_sale: Migration to 8.0

* Migration and enhancement of all modules involved in SEPA
* Remove duplicated files
* Check if payment mode is set in the propagation
* Clean files
* account_payment_sale_stock: Split for removing stock dependency on sale
* Remove old translations templates
* Reorder on folder structure
* Translation template files
* Translations to spanish
* Use sale_ok and purchase_ok instead of tweaking payment type
* Add sale_ok and purchase_ok filters in partner/sale/purchase form views
* Use widget=selection for payment_mode_id fields
* Update demo data

alexis-via committed:
[MIG] account_payment_sale: Migration to v9

Commit:
Port almost all modules to v10 (#305)

Port almost all modules to v10

* Update to EPC Rulebook v9.2 that start to apply on 2016-11-20 (bug #300)

cubells committed:
[IMP] account_payment_sale: add payment mode to sale report

Commit:
[MIG] account_payment_sale: Migrate to 11.0

Commit:
[IMP] account_payment_sale: Add tests to increase coverage

mourad-ehm committed:
[MIG] account_payment_sale: migrate to 12.0

pedrobaeza committed:
[MIG] account_payment_sale: Adapt tests to v12

lmignon committed:
[FIX] account_payment_sale: Propagate payment_mode to invoices created from a call to sale.action_invoice_create
Into the normal process of the invoice creation from a sale order by odoo, the onchange_partner is called. In some case, this call will reset the specific payment_mode and/or partner_bank_id set on the sale_order.
To fix this issue, we must override the _finalize_invoices method, take a snapshot of these values before the call to super and restore these values if these are changed.
fixes #536

rven committed:
[IMP] account_payment_sale: black, isort

carlosdauden committed:
[FIX] account_payment_sale: Compatibility with account_banking_sepa_direct_debit

Steps to reproduce the problem:

- Set a payment mode with fixed journal and set the journal with bank account A.
- Create a sales order and put bank account B.
- Create invoice.
- Bank account in invoice is A instead of B.

rven committed:
[MIG] account_payment_sale: Migration to 13.0

OCA-git-bot committed:
pre-commit update

alexis-via committed:
[MIG] account_payment_sale to 14.0

pedrobaeza committed:
[FIX] account_payment_sale: Adapt tests to 18489012

The expected one is to not have partner bank set, as we don't want the
invoice is printed with such bank account forced.

manuelregidor committed:
[MIG]account_payment_sale: Migration to 15.0

SodexisTeam committed:
[MIG] account_payment_sale: Migration to 16.0

alexis-via committed:
account_payment_sale: payment_mode_id in sale.report

alexis-via committed:
account_payment_sale: remove unneeded code

Remove inherit of sale.advance.payment.inv: no need to inherit
_prepare_invoice_values() because it calls _prepare_invoice() from the
sale.order object

josep-tecnativa committed:
[IMP] account_payment_sale: test performance improvement
- Create setUpClass method
- Include context keys for avoiding mail operations overhead.

ramiadavid committed:
[IMP] account_payment_sale: pre-commit auto fixes

ramiadavid committed:
[MIG] account_payment_sale: Migration to 17.0

Commit:
[UPD] Update account_payment_sale.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: bank-payment-17.0/bank-payment-17.0-account_payment_sale
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-17-0/bank-payment-17-0-account_payment_sale/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: bank-payment-17.0/bank-payment-17.0-account_payment_sale
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-17-0/bank-payment-17-0-account_payment_sale/it/

ramiadavid committed:
[IMP] account_banking_mandate: pre-commit auto fixes

rousseldenis committed:
[IMP] account_payment_sale: Add payment mode in invoicing grouping criteria

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (4 of 4 strings)

Translation: bank-payment-17.0/bank-payment-17.0-account_payment_sale
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-17-0/bank-payment-17-0-account_payment_sale/pt_BR/

santiagopim committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (4 of 4 strings)

Translation: bank-payment-17.0/bank-payment-17.0-account_payment_sale
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-17-0/bank-payment-17-0-account_payment_sale/ca/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (4 of 4 strings)

Translation: bank-payment-17.0/bank-payment-17.0-account_payment_sale
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-17-0/bank-payment-17-0-account_payment_sale/sv/

pedrobaeza committed:
[FIX] account_payment_sale: Remove XML header in HTML

For avoiding the warning.

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

alexis-via committed:
[IMP] account_payment_sale: pre-commit auto fixes

alexis-via committed:
[MIG] account_payment_sale: migrate from 17 to 18

Now depend on account_payment_base_oca

ioans73 commented:
Hi @alexis-via 
Do you plan to continue working on this PR?

leemannd commented on account_payment_sale/models/sale_order.py:
```diff
@@ -0,0 +1,56 @@
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    payment_method_line_id = fields.Many2one(
```
The change of model would need a computation script

alexis-via replied:
Of course !

alexis-via committed:
[DON'T MERGE] add test-requirements.txt

eugenios73 commented:
Hi @alexis-via! 
Do you have plan to continue working in this pull request at short time?

alexis-via commented:
@eugenios73 Yep. It will depend on the result of the debate that you can follow here: https://odoo-community.org/groups/contributors-15/contributors-2029602

pedrobaeza commented:
I think the debate is clear already: what is merged here in this repo is the plain migration, and in fact, the module is already merged: https://github.com/OCA/bank-payment/tree/18.0/account_payment_sale.

alexis-via commented:
Arf ! No comment...

alexis-via committed:
Rename module to account_payment_base_oca_sale

Update README
Update website key in manifest to point to bank-payment-alternative

alexis-via committed:
[DON'T MERGE] update test-requirements.txt

alexis-via commented:
PR moved to https://github.com/OCA/bank-payment-alternative/pull/12
